collector/systemd_linux.go

```diff
@@ -132,6 +137,9 @@ func NewSystemdCollector(logger *slog.Logger) (Collector, error) {
 	systemdVersionDesc := prometheus.NewDesc(
 		prometheus.BuildFQName(namespace, subsystem, "version"),
 		"Detected systemd version", []string{"version"}, nil)
+	virtualizationDesc := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystem, "virtualization"),
+		"Detected virtualization technology", []string{"type"}, nil)
```

Comment:
Two nits:
* This should be an `_info` metric to match the naming conventions.
* We try and avoid `type` as a label as it's too generic.

```suggestion
		prometheus.BuildFQName(namespace, subsystem, "virtualization_info"),
		"Detected virtualization technology", []string{"virtualization_type"}, nil)
```